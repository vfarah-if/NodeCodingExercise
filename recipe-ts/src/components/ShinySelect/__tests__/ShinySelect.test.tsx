import React from 'react';
import '@testing-library/jest-dom';
import { render, fireEvent, screen } from '@testing-library/react';

import {
	OptionGroups,
	OptionsAndGroupsMixed,
	SimpleOptions,
	ValuesOnly,
} from '../ShinySelect.stories';

describe('ShinySelect', () => {
	test('should create option groups', () => {
		const { container } = render(
			<OptionGroups
				label={OptionGroups.args?.label}
                optionsList={OptionGroups.args?.optionsList || []}
                value={OptionGroups.args?.value || ''}
			/>
		);

		expect(container).toMatchSnapshot();
    });
    
    test('should create options and groups mixed', () => {
		const { container } = render(
			<OptionsAndGroupsMixed
				label={OptionsAndGroupsMixed.args?.label}
                optionsList={OptionsAndGroupsMixed.args?.optionsList || []}
                value={OptionsAndGroupsMixed.args?.value || ''}
			/>
		);

		expect(container).toMatchSnapshot();
    });
    
    test('should create options only', () => {
		const { container } = render(
			<SimpleOptions
				label={SimpleOptions.args?.label}
                optionsList={SimpleOptions.args?.optionsList || []}
                value={SimpleOptions.args?.value || ''}
			/>
		);

		expect(container).toMatchSnapshot();
    });
    
    test('should create values only', () => {
		const { container } = render(
			<ValuesOnly
				label={ValuesOnly.args?.label}
                optionsList={ValuesOnly.args?.optionsList || []}
                value={ValuesOnly.args?.value || ''}
			/>
		);

		expect(container).toMatchSnapshot();
	});
});
