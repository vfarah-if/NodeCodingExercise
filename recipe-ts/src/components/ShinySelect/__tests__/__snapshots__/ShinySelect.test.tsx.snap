// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShinySelect should create option groups 1`] = `
<div>
  <label>
    Select a car
    <select
      class="select-css"
    >
      <optgroup
        label="Swedish Cars"
      >
        <option
          value="volvo"
        >
          Volvo
        </option>
        <option
          value="saab"
        >
          Saab
        </option>
      </optgroup>
      <optgroup
        label="German Cars"
      >
        <option
          value="vw"
        >
          VW
        </option>
        <option
          value="mercedes"
        >
          Merc
        </option>
        <option
          value="audi"
        >
          Audi
        </option>
      </optgroup>
    </select>
  </label>
</div>
`;

exports[`ShinySelect should create options and groups mixed 1`] = `
<div>
  <label>
    Select a car
    <select
      class="select-css"
    >
      <option
        value=""
      >
        Select a car
      </option>
      <optgroup
        label="Swedish Cars"
      >
        <option
          value="volvo"
        >
          Volvo
        </option>
        <option
          value="saab"
        >
          Saab
        </option>
      </optgroup>
      <optgroup
        label="German Cars"
      >
        <option
          value="vw"
        >
          VW
        </option>
        <option
          value="mercedes"
        >
          Merc
        </option>
        <option
          value="audi"
        >
          Audi
        </option>
      </optgroup>
    </select>
  </label>
</div>
`;

exports[`ShinySelect should create options only 1`] = `
<div>
  <label>
    Choose a Mars rover
    <select
      class="select-css"
    >
      <option
        value=""
      >
        Select your option
      </option>
      <option
        value="curiosity"
      >
        Curiosity
      </option>
      <option
        value="spirit"
      >
        Spirit
      </option>
      <option
        value="opportunity"
      >
        Opportunity
      </option>
    </select>
  </label>
</div>
`;

exports[`ShinySelect should create values only 1`] = `
<div>
  <label>
    <select
      class="select-css"
    >
      <option
        value=""
      >
        Select a car
      </option>
      <option
        value="volvo"
      >
        volvo
      </option>
      <option
        value="saab"
      >
        saab
      </option>
      <option
        value="vw"
      >
        vw
      </option>
      <option
        value="mercedes"
      >
        mercedes
      </option>
      <option
        value="audi"
      >
        audi
      </option>
    </select>
  </label>
</div>
`;
